src/background.ts

```diff
@@ -0,0 +1,284 @@
+  "appId": "1:982322764946:web:f9b6aea488cebde47ada4b",
+  "functionsHost": "https://us-central1-moz-fx-data-rally-w-prod-dfa4.cloudfunctions.net"
+};
+
+// Overrides for dev mode - use local emulators with "exampleStudy1" as study ID.
+if (enableEmulatorMode) {
+  studyId = "exampleStudy1";
+  rallySite = "http://localhost:3000";
+  firebaseConfig = {
+    "apiKey": "abc123",
+    "authDomain": "demo-rally.firebaseapp.com",
+    "projectId": "demo-rally",
+    "storageBucket": "demo-rally.appspot.com",
+    "messagingSenderId": "abc123",
+    "appId": "1:123:web:abc123",
+    "functionsHost": "http://localhost:5001"
+  }
+}
+
+// This function will be called when the study state changes. By default,
+// a study starts "paused". If a user opts-in to a particular study, then the
+// state will change to "started".
+//
+// The study state may change at any time (for example, the server may choose to pause a particular study).
+// Studies should stop data collection and try to unload as much as possible when in "paused" state.
+async function stateChangeCallback(newState) {
+  switch (newState) {
+    case (RunStates.Running):
+      // The all-0 Rally ID indicates developer mode, in case data is accidentally sent.
+      let rallyId = enableDevMode ? "00000000-0000-0000-0000-000000000000" : rally.rallyId;
+
+      // The all-1 Rally ID means that there was an error with the Rally ID.
+      if (!rallyId) {
+        rallyId = "11111111-1111-1111-1111-111111111111";
+      }
+      console.info(`Study running with Rally ID: ${rallyId}`);
+
+      const storage = await browser.storage.local.get("enrolled");
+      if (storage.enrolled !== true) {
+        console.info("Recording enrollment.");
+        rallyManagementMetrics.id.set(rallyId);
+        examplePings.studyEnrollment.submit();
+
+        browser.storage.local.set({
+          enrolled: true,
+        });
+      }
+
+      // The Rally API has been initialized.
+      // Initialize the study and start it.
+
+      // Example: initialize the example module.
+      exampleModuleInitialize();
+      await browser.storage.local.set({ "state": RunStates.Running });
+
+
+      // Example: set a listener for WebScience page navigation events on
+      // http://localhost/* pages. Note that the manifest origin
+      // permissions currently only include http://localhost/*. You should
+      // update the manifest permissions as needed for your study.
+
+      this.pageDataListener = async (pageData) => {
+        console.log(`WebScience page navigation event fired with page data:`, pageData);
+        userJourney.pageId.set(pageData.pageId);
+        // Round attention duration up to 1ms, Glean will balk if it is lower than this.
+        if (pageData.attentionDuration < 1) {
+          userJourney.attentionDuration.set(1);
+        } else {
+          userJourney.attentionDuration.set(pageData.attentionDuration);
+
+        }
+        userJourney.audioDuration.set(pageData.audioDuration);
```

Comment:
as a side note; do we want to set attentionAndAudioDuration here as well? The Glean metric is defined;; I'm not exactly sure how it's useful though...

Reply:
The only place it's really documented is https://mozilla-rally.github.io/web-science/jsdocs/module-pageNavigation.onPageData.html - I think the key is that it's how long the user spent paying attention when there was also audio playing at the same time, the regular audio metric can be in a background tab etc.

Reply:
ah right ok. I'm thinking it could be good to add it for completion's sake? Since we have the rest of the pageNavigation fields going in.

Reply:
Yep, sorry I misread your original comment here, that was not left out on purpose :) thanks!

Reply:
oh yep I see how that was confusing haha

Reply:
Nice catch, thanks!